src/Sylius/Bundle/CartBundle/Provider/SessionCartProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CartBundle\Provider;
+
+use Sylius\Component\Cart\Provider\CartProviderInterface;
+use Sylius\Component\Cart\Repository\CartRepositoryInterface;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+class SessionCartProvider implements CartProviderInterface
+{
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var string
+     */
+    private $sessionKeyName;
+
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * @param SessionInterface $session
+     * @param string $sessionKeyName
+     * @param CartRepositoryInterface $cartRepository
+     */
+    public function __construct(
+        SessionInterface $session,
+        $sessionKeyName,
```

Comment:
We can extract `CartStorageInterface` with methods `set(CartInterface $cart): void` and `get(): CartInterface` so we won't have to pass these two separately and this provider can become storage type independent.